app/app.go

```diff
@@ -717,6 +726,9 @@ func (d *Deployments) CreateDeployment(ctx context.Context,
 				break
 			}
 			constructor.Devices = append(constructor.Devices, inventoryDevicesToDevicesIds(devices)...)
+			if len(constructor.Devices) == count {
+				break
+			}
```

Comment:
this is just an extra condition in case the first search returned all the devices we need.

Reply:
yes, but it also prevents an extra call to the inventory if we have all the devices we expect, already.